File: C/Algorithms/Math/greatestCommonDivisor.c
/**
 * Calculates the greatest common divisor of two numbers using Euclidean algorithm
 *
 * @param int firstNumber
 * @param int secondNumber
 * @return int
 */
 int greatestCommonDivisor(int firstNumber, int secondNumber) {
    while (firstNumber != secondNumber) {
        if (firstNumber > secondNumber) {
            firstNumber -= secondNumber;
        } else {
            secondNumber -= firstNumber;
        }
    }

    return firstNumber;
}

File: Java/Algorithms/Recursion/GreatestCommonDivisor.java
/**
 * Class GreatestCommonDivisor
 */
public class GreatestCommonDivisor {
    /**
     * Calculates the greatest common divisor of two numbers using recursion
     *
     * @param int firstNumber a given integer
     * @param int secondNumber a given integer
     * @return int
     */
     public int greatestCommonDivisor(int firstNumber, int secondNumber) {
        if (firstNumber == 0) {
            return secondNumber;
        }

        return greatestCommonDivisor(secondNumber % firstNumber, firstNumber);
    }
}
